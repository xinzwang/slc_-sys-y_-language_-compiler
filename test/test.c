// int getint( );
// void putint(int x);

/*
const int ca = 0;
const int cb[5] = {1, 2, 3, 4, 5};

int a = 6;
int b[5] = {11, 8, 10, 7, 9};

int fun( ) {
	int t = 0;
	if (t > 0) {
		t = cb[t - ca + a];
		b[t] = getint( );
		t = t - b[t];
		return t * fun(t - 1);
	} else {
		return t;
	}
}

int main( ) {
	int k, i;
	int arr[5];

	a = getint( );
	arr[0] = getint( );
	arr[1] = getint( );
	arr[2] = getint( );

	arr[3] = getint( );
	arr[4] = getint( );

	i = arr[1];
	arr[1] = arr[i] - (10 + a) * a - 8;
	i = 0;

	while (i < 5) {
		if (i % 2 == 0) {
			k = k + cb[i];
		} else {
			k = k + b[i];
			// b[k] = getint( );
		}
		k = k + arr[i];
		if (k > 3) {
			k = k - a;
		} else {
			k = k - ca;
		}
		i = i + 1;
	}

	k = fun(k);
	putint(k);
}
*/

int b = 0;
int cc[3] = {5, 5, 5};
int main()
{
	int a = 0;
	int m = 0;
	int n = 100;
	int j = 11;
	int k = 100;
	a = m + n;
	return m;
}