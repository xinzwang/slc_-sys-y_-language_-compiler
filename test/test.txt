-------------------------  IR  -------------------------

Global const list:
    <const_int>  name:ca  addr:@ca  size:4  val:123

Global var list:
    <global_var_int>  name:a  addr:@a  size:4  val:1

Function:  type:void  name:fun  fparams:(<type:int name:a><type:int name:b>)
block_1:


Function:  type:int  name:main  fparams:()
block_1:
    [Alloc]  dest:%a  size:4
    [Alloc]  dest:%b  size:4
    [Store]  dest:%b  v:0
    [Alloc]  dest:%c  size:4
    [Store]  dest:%c  v:0
    [Add]  dest:%1  v1:%b  v2:%c
    [Store]  dest:%a  v:%1
    [Call]  dest:%2  func:fun  rparms:(<v:%b><v:%c>)
    [GEP]  dest:%3  v:%a  inbounds:(1)  inrange:(0)
    [Load]  dest:%4  v:%b
    [Store]  dest:%3  v:%4
