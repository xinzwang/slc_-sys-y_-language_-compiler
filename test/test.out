    .arch armv7-a
    .file "./test"
    .text
    .global main
    .arm
    .fpu neon
    .global b
    .global cc
.global_vars:
    .long b
    .long cc
main:
    @ stack_size=128
main:
    push {fp, lr}
    add    fp, sp, #4
    sub    sp, sp, #232
.main_block_1
    @    %a:-1 
    @    %a:-1 
    mov    r0, #0
    str    r0, [fp, #-4]
    @    %m:-1 
    @    %m:-1 
    mov    r0, #0
    str    r0, [fp, #-8]
    @    %n:-1 
    @    %n:-1 
    mov    r0, #0
    str    r0, [fp, #-12]
    @    %j:-1 
    @    %j:-1 
    mov    r0, #0
    str    r0, [fp, #-16]
    @    %k:-1 
    @    %k:-1 
    mov    r0, #0
    str    r0, [fp, #-20]
    @    %1:12 %m:-1 %n:-1 
    ldr    r0, [fp, #-8]
    ldr    r1, [fp, #-12]
    add    r12, r0, r1
    @    %a:-1 %1:12 
    str    r12, [fp, #-4]
    @    %m:-1 
    ldr    r1, [sp, #-8]
    mov     r0, r1
    sub    sp, fp, #4
    pop    {fp, pc}
    .data
b:
    .long 0
cc:
    .long 5
    .long 5
    .long 5
